File: Back/Python_Basic/app/routes/order.py
from flask import Blueprint, request
from app.services.order_service import OrderService
from app.common.base_response import BaseResponse

order_bp = Blueprint('orders', __name__, url_prefix='/api/orders')
order_service = OrderService()


@order_bp.route('', methods=['POST'])
def create_order():
    data = request.get_json()
    order = order_service.create_order(data)
    return BaseResponse.success("订单创建成功", order)


@order_bp.route('/<int:order_id>/status', methods=['PUT'])
def update_order_status(order_id):
    data = request.get_json()
    new_status = data.get('status')
    updated_order = order_service.update_order_status(order_id, new_status)
    return BaseResponse.success("订单更新完成", updated_order)


@order_bp.route('/<int:order_id>', methods=['GET'])
def get_order_detail(order_id):
    detail = order_service.get_order_detail(order_id)
    return BaseResponse.success("订单信息获取成功", detail)


@order_bp.route('/buyer/<int:buyer_id>', methods=['GET'])
def get_buyer_orders(buyer_id):
    status = request.args.get('status')
    orders = order_service.get_buyer_orders(buyer_id, status)
    return BaseResponse.success("获取买家订单列表成功", orders)


@order_bp.route('/seller/<int:seller_id>', methods=['GET'])
def get_seller_orders(seller_id):
    status = request.args.get('status')
    orders = order_service.get_seller_orders(seller_id, status)
    return BaseResponse.success("获取卖家订单列表成功", orders)


@order_bp.route('/<int:order_id>', methods=['DELETE'])
def delete_order(order_id):
    order_service.delete_order(order_id)
    return BaseResponse.success("订单删除成功")

File: Front/secondhand-trading-frontend/src/api/order.js
import axios from 'axios'
import { API_CONFIG } from '../config/api.js'

const BASE_URL = API_CONFIG.BASE_URL

// 创建新订单
export function createOrder(data) {
  return axios.post(`${BASE_URL}/api/orders`, data)
}

// 更新订单状态
export function updateOrderStatus(orderId, data) {
  return axios.put(`${BASE_URL}/api/orders/${orderId}/status`, data)
}

// 获取订单详情
export function getOrderDetail(orderId) {
  return axios.get(`${BASE_URL}/api/orders/${orderId}`)
}

// 获取买家订单列表
export function getBuyerOrders(buyerId, status) {
  console.log('调用买家订单API, buyerId:', buyerId, 'status:', status)
  return axios.get(`${BASE_URL}/api/orders/buyer/${buyerId}`, {
    params: { status }
  })
}

// 获取卖家订单列表
export function getSellerOrders(sellerId, status) {
  console.log('调用卖家订单API, sellerId:', sellerId, 'status:', status)
  return axios.get(`${BASE_URL}/api/orders/seller/${sellerId}`, {
    params: { status }
  })
}

// 删除订单
export function deleteOrder(orderId) {
  return axios.delete(`${BASE_URL}/api/orders/${orderId}`, {
    data: orderId
  })
}

export function orderApi(){
  return
}